domain/src/main/java/com/stop/domain/usecase/route/GetLastTransportTimeUseCaseImpl.kt

```diff
@@ -16,132 +18,358 @@ internal class GetLastTransportTimeUseCaseImpl @Inject constructor(
 
-    private suspend fun getBusLastTransportTime(transportIdRequest: TransportIdRequest): Int {
-        when (transportIdRequest.area) {
-            Area.GYEONGGI -> {
-                TODO()
-            }
-            Area.SEOUL -> {
-                val lastTime = routeRepository.getSeoulBusLastTime(
-                    transportIdRequest.stationId,
-                    transportIdRequest.lineId
-                ).toInt()
-
-                if (lastTime < MID_NIGHT) {
-                    return lastTime + MID_NIGHT
+    private fun checkInnerOrOuter(
+        stationType: Int,
+        startStationIndex: Int,
+        endStationIndex: Int,
+        stationsOfLine: List<Station>,
+    ): SubwayCircleType {
+        return if (stationType == 2) {
+            if (startStationIndex < endStationIndex) {
+                if (stationsOfLine[startStationIndex].frCode.contains("211-") // 성수 ~ 신설동 예외처리
+                    || stationsOfLine[endStationIndex].frCode.contains("211-")
+                ) {
+                    SubwayCircleType.OUTER
+                } else {
+                    SubwayCircleType.INNER
+                }
+            } else {
+                if (startStationIndex - endStationIndex >= EMPIRICAL_DISTINCTION) {
+                    SubwayCircleType.INNER
+                } else {
+                    SubwayCircleType.OUTER
                 }
-                return lastTime
             }
-            Area.UN_SUPPORT_AREA -> {
-                TODO()
+        } else {
+            if (startStationIndex < endStationIndex) {
+                SubwayCircleType.OUTER
+            } else {
+                SubwayCircleType.INNER
             }
         }
     }
 
+    private fun checkStationCase(
+        stationType: Int,
+        subwayCircleType: SubwayCircleType,
+        startIndex: Int,
+        endIndex: Int,
+        stationsOfLine: List<Station>,
+    ): Boolean {
+        if (stationType != 2) {
+            return false
+        }
+
+        if (subwayCircleType == SubwayCircleType.OUTER) {
+            if (startIndex < endIndex) {
+                if (stationsOfLine[startIndex].frCode.contains("211-") // 성수 ~ 신설동 예외처리
+                    || stationsOfLine[endIndex].frCode.contains("211-")
+                ) {
+                    return false
+                }
+                return true
+            }
+            return false
+        }
+        return startIndex > endIndex
+    }
 
-    private suspend fun getSeoulBusLineId(transportIdRequest: TransportIdRequest): TransportIdRequest {
-        val busName = transportIdRequest.lineName.split(":")[1]
-        val line =
-            routeRepository.getSeoulBusLine(transportIdRequest.stationId).lineIdMsgBody.busLines.firstOrNull {
-                it.busLineName.contains(busName)
-            } ?: throw NoAppropriateDataException(NO_BUS_LINE_ID)
+    /**
+     * 요일 별로 막차 시간이 다르기 때문에, 앱을 실행하는 오늘의 요일도 받아야 한다.
+     */
+    private fun whaWeekToday(): WeekType {
```

Comment:
```suggestion
    private fun getDayOfWeek(): WeekType {
```
wha는 오타인가요? 오늘의 요일이라는 뜻의 day of week를 쓰면 좋을 것 같습니다.

Reply:
[4d6e92b](https://github.com/boostcampwm-2022/android10-PlzStop/pull/57/commits/4d6e92b0f8040553b119266ae1126d96e9afa66c)